### ISSUE TITLE
Add support for sets

### ISSUE THREAD

**Issue by NameNotFinal** [rorre/Yippi#106]
Currently the API documentation on the site says nothing about sets, but there exists some useful endpoints that aren't supported by Yippi.

The endpoints I know about are:
- post_sets.json
- post_sets/{ID}.json
- post_sets/{ID}/add_posts.json
- post_sets/{ID}/remove_posts.json

Seems like the add/remove enpoints takes an object with a `post_ids` key, that has a list of IDs to add/remove to the set.

This might be useful to look at:
https://github.com/zwagoth/e621ng/blob/master/app/javascript/src/javascripts/post_sets.js

**Comment by Deer-Spangle** [comment]
To be honest, I think this library is pretty much unmaintained now. I don't have a tonne of time to poke at it myself, between other projects.


Though, I have had my eyes on this other lib: https://github.com/PatriotRossii/e621-py which seems to have pretty thorough coverage of the API, even undocumented endpoints. Hope that might help

**Comment by rorre** [comment]
Yes, I have not been able to touch at this thing for a long time now (work and uni has been kicking my fluff especially), and that I am also quite blocked with how to mock uploads etc. I have not looked at the wiki as well, but from last time it was kind of lacking.

The library linked looks rather promising with modern usage of Python's typing, in the meantime while I am unable to maintain this repo, if that library helps more feel free to use it :)
